### PR TITLE
Deprecate initialization delay

### PR DESCRIPTION
**Pull request by kdnilsen**
Automatically calculate the initialization delay rather than requiring users to guess and specify.  The original InitializationDelay command-line argument is still supported for compatibility with existing scripts, but the value is ignored.

Additionally, fix up a few comments and remove dependencies on deprecated new Long() constructor.

### CODE REVIEW

**Comment by earthling-amzn** [comment]
Is there any notion of initialization time out now? or will it just take as long as it takes?

**Comment by kdnilsen** [comment]
There is no enforcement of a timeout on initialization.  After almost all of the shared data structures and Threads are allocated, we check the current time.  We have not yet started the customer or server threads.  We "assume" it will take 4 ms per thread to start them all up.  (measurements showed it takes approximately 2 ms per thread on my cloud desktop). We set the simulation start time to the current time plus the anticipated time to startup all threads.

If it takes longer than our assumed 4 ms per thread, we'll issue a warning message, but will not abort the simulation.

File: Extremem/src/main/java/com/amazon/corretto/benchmark/extremem/Configuration.java [hunk 1]
```diff
@@ -183,11 +181,11 @@ void initialize(ExtrememThread t) {
                    2 * Util.SizeOfFloat + 2 * Util.SizeOfBoolean);
 
     // Account for 11 reference fields: args, dictionary,
```
Comment (earthling-amzn) [inline Extremem/src/main/java/com/amazon/corretto/benchmark/extremem/Configuration.java:183]: Account for 10 ...

**Review by earthling-amzn — APPROVED** [review]

File: Extremem/src/main/java/com/amazon/corretto/benchmark/extremem/Configuration.java (hunk 1 above)
Comment (kdnilsen) [reply Extremem/src/main/java/com/amazon/corretto/benchmark/extremem/Configuration.java:183]: Good catch.  Thanks.  Fixed.

**Pull request merged**